sdk/src/main/scala/com/horizen/block/MainchainTransaction.scala

```diff
@@ -1,19 +1,19 @@
 package com.horizen.block
 
-import com.horizen.utils.{ByteArrayWrapper, BytesUtils, Utils, VarInt}
-import scala.util.Try
+import com.horizen.utils.{ByteArrayWrapper, BytesUtils, Utils, VarInt, _}
 
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
 
 class MainchainTransaction(
                           transactionsBytes: Array[Byte],
                           version: Int,
-                          crosschainOutputsMap: Map[ByteArrayWrapper, Seq[MainchainTxCrosschainOutput]]
+                          crosschainOutputsMap: Map[ByteArrayWrapper, Seq[MainchainTxCrosschainOutput]],
+                          val hash: Array[Byte]
```

Comment:
I would prefer to avoid passing `hash` as external argument. `transactionsBytes` is enough.
I understand, that it will allow, to save one hashing operation, but logically it's a source of mistakes by passing inconsistent data and doesn't encapsulate the logic inside MainchainTransaction

Reply:
changed to calculate inside transaction